viz/SonarVisualization.cpp

```diff
@@ -0,0 +1,174 @@
+#include "SonarVisualization.hpp"
+#include <osg/Geode>
+#include <base/Eigen.hpp>
+
+namespace vizkit3d
+{
+
+SonarVisualization::SonarVisualization():
+    fan_opening(0),
+    new_sonar_scan(false),
+    full_scan(false)
+{}
+
+osg::ref_ptr< osg::Node > SonarVisualization::createMainNode()
+{
+    osg::ref_ptr<osg::Group> mainNode = new osg::Group();
+    pointGeom = new osg::Geometry;
+    pointsOSG = new osg::Vec3Array;
+    pointGeom->setVertexArray(pointsOSG);
+    color = new osg::Vec4Array;
+    pointGeom->setColorArray(color);
+    pointGeom->setColorBinding(osg::Geometry::BIND_PER_VERTEX);
+    pointGeom->getOrCreateStateSet()->setMode(GL_LIGHTING, osg::StateAttribute::OFF); 
+    drawArrays = new osg::DrawArrays( osg::PrimitiveSet::TRIANGLES, 0, pointsOSG->size() );
+    pointGeom->addPrimitiveSet(drawArrays.get());
+    
+    osg::ref_ptr<osg::Geode> geode = new osg::Geode;
+    geode->addDrawable(pointGeom.get());
+    mainNode->addChild(geode);
+    return mainNode;
+}
+
+double SonarVisualization::angleDifference(double a,double b)
```

Comment:
1. You don't seem to be using this
2. If you need something like that, just use [base::Angle](https://github.com/rock-core/base-types/blob/master/src/Angle.hpp)